qiskit_nature/properties/electronic_structure/angular_momentum.py

```diff
@@ -10,32 +10,80 @@
 # that they have been altered from the originals.
 
-""" Calculator of 1- and 2-body integrals for a total angular momentum."""
+"""The AngularMomentum property."""
+
+from typing import cast, List, Tuple
 
 import itertools
-from typing import Tuple, List
 
 import numpy as np
 
+from qiskit_nature.drivers.second_quantization import QMolecule
+from qiskit_nature.operators.second_quantization import FermionicOp
+from qiskit_nature.results import EigenstateResult
 
-def calc_total_ang_momentum_ints(num_modes: int) -> Tuple[np.ndarray, np.ndarray]:
-    """
-    Calculates 1- and 2-body integrals for a total angular momentum.
+from .bases import ElectronicBasis
+from .integrals import (
+    OneBodyElectronicIntegrals,
+    TwoBodyElectronicIntegrals,
+)
+from ..property import DriverResult, ElectronicDriverResult, Property
 
-    Args:
-        num_modes (int): Number of modes.
 
-    Returns:
-        Tuple(numpy.ndarray, numpy.ndarray): Tuple of 1- and 2-body integrals for a total angular
-        momentum.
-    """
-    x_h1, x_h2 = _calc_s_x_squared_ints(num_modes)
-    y_h1, y_h2 = _calc_s_y_squared_ints(num_modes)
-    z_h1, z_h2 = _calc_s_z_squared_ints(num_modes)
-    h_1 = x_h1 + y_h1 + z_h1
-    h_2 = x_h2 + y_h2 + z_h2
+class AngularMomentum(Property):
+    """The AngularMomentum property."""
 
-    return h_1, h_2
+    def __init__(self, num_spin_orbitals: int):
+        """
+        Args:
+            num_spin_orbitals: the number of spin orbitals in the system.
+        """
+        super().__init__(self.__class__.__name__)
+        self._num_spin_orbitals = num_spin_orbitals
+
+    @classmethod
+    def from_driver_result(cls, result: DriverResult) -> "AngularMomentum":
+        """Construct an AngularMomentum instance from a QMolecule.
+
+        Args:
+            result: the driver result from which to extract the raw data. For this property, a
+                QMolecule is required!
+
+        Returns:
+            An instance of this property.
+
+        Raises:
+            QiskitNatureError: if a WatsonHamiltonian is provided.
+        """
+        cls._validate_input_type(result, ElectronicDriverResult)
+
+        qmol = cast(QMolecule, result)
+
+        return cls(
+            qmol.num_molecular_orbitals * 2,
+        )
+
+    def second_q_ops(self) -> List[FermionicOp]:
+        """Returns a list containing the angular momentum operator."""
+        x_h1, x_h2 = _calc_s_x_squared_ints(self._num_spin_orbitals)
+        y_h1, y_h2 = _calc_s_y_squared_ints(self._num_spin_orbitals)
+        z_h1, z_h2 = _calc_s_z_squared_ints(self._num_spin_orbitals)
+        h_1 = x_h1 + y_h1 + z_h1
+        h_2 = x_h2 + y_h2 + z_h2
+
+        h1_ints = OneBodyElectronicIntegrals(ElectronicBasis.SO, h_1)
+        h2_ints = TwoBodyElectronicIntegrals(ElectronicBasis.SO, h_2)
+        return [(h1_ints.to_second_q_op() + h2_ints.to_second_q_op()).reduce()]
+
+    def interpret(self, result: EigenstateResult) -> None:
```

Comment:
Since the implementation in the future will likely look similar to this:
```python
class ElectronicStructureProblem:
    def interpret(self, result: EigenstateResult) -> ElectronicStructureResult:
        new_result = ElectronicStructureResult(result)
        for prop in self.properties:
            prop.interpret(new_result)
```
I was thinking that these `interpret` methods add meaning to the `result`-object **in-place**. If this is unwanted we can change this when addressing #243.

Reply:
I removed the `interpret` method stubs from this PR